src/lib/syncPagePath.ts

```diff
@@ -208,6 +208,23 @@ export default async function syncPagePath(this: CommandsThatSyncMarkdown) {
     throw err;
   });
 
+  // check whether or not the project has bidirection syncing enabled
+  // if it is, validations must be skipped to prevent frontmatter from being overwritten
+  const headers = new Headers({ authorization: `Bearer ${key}` });
+  const projectMetadata: ProjectRepresentation = await this.readmeAPIFetch('/projects/me', {
+    method: 'GET',
+    headers,
+  }).then(res => {
+    return this.handleAPIRes(res);
+  });
+
+  const biDiConnection = projectMetadata.data.git?.connection;
+  if (biDiConnection && !skipValidation) {
+    throw new Error(
+      "Bi-directional syncing is enabled for this project. Uploading these docs will overwrite what's currently synced from Git. To proceed with uploading via `rdme`, please use the `--skip-validation` flag.",
+    );
+  }
+
   if (skipValidation) {
```

Comment:
thoughts on making it so this warning is only shown if bidi is **not** enabled? or maybe we have a separate warning for those users?

```suggestion
  if (skipValidation && !bidiConnection) {
```

Reply:
If we had a separate warning for `skipValidation` + bidi, what would we wanna say? Should we repeat the warning from the bidi without `skipValidation` error? (ie "Bi-directional syncing is enabled for this project. Uploading these docs will overwrite what's currently synced from Git.")

Reply:
yep i think that's perfect!